### PR TITLE
 Windows: native support

### LINKED ISSUE #841: Sockets are always beeing closed
In `sendrecv.py`, we are always closing the sockets after use. This can result in strange behaviors:

```
>>> a = conf.L2socket()
>>> send(IP(), socket=a) # Works but closes socket at the end
>>> send(IP(), socket=a) # Fails
WindowsError: exception.....
```

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- implements what was discussed in https://github.com/secdev/scapy/issues/1747, knowingly L3 sockets that work without Winpcap. Fixes https://github.com/secdev/scapy/issues/1747
- added a ping as test which is currently working (but requires complicated metrics explained in native.py), for those sockets
- Added `conf.L3socket6` which sort of already existed, but now is clarified
- fixes #841
- implements `get_windows_if_list` and `read_routes(6)` in C/ctypes. It’s much faster and reliable. Tested on W10, W7 and WXP
- this speeds up Windows booting (fixes https://github.com/secdev/scapy/issues/1699)
- removes `POWERSHELL_PROCESS` and all its mechanics, replaced by the previous + cleanup
- load interfaces independently from winpcapy, and do not destroy others but mark them as invalid.
- rename `mock_windows.uts` -> `windows.uts` for consistency with `linux.uts` and `osx.uts`

## Benchmark: loading time (WINDOWS)
```python
import time
i = time.time()
from scapy.all import *
print(time.time()-i)
```
| State | Python 2.7.13 | Python 3.7.2 |
|---|---|---|
| secdev/master | 3.68s | 3.27s |
| This PR |  1.24s | 1.07s |

Notes:
- the socket must be configured to receive specifically IP or IPv6, while being setup.
- **It's incredibly hard to setup tests on C calls :/ They are no tests right now. I have tested it on Windows 10 and XP**

### CODE REVIEW

File: scapy/arch/windows/native.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            self.ins = socket.socket(socket.AF_INET,
+                                     socket.SOCK_RAW,
+                                     socket.IPPROTO_IP)
+            self.outs = socket.socket(socket.AF_INET,
+                                      socket.SOCK_RAW,
+                                      socket.IPPROTO_IP)
+        except OSError as e:
+            if e.errno == 10013:
+                raise OSError("Windows L3 Raw sockets are only "
+                              "useable as administrator !")
```
Comment (p-l-) [inline scapy/arch/windows/native.py]: useable => usable (codespell)

**Comment by p-l-** [comment]
You also have flake8 errors.

I think this is going to be a great PR!

File: scapy/arch/windows/native.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            raise
+        self.ins.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        self.outs.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        # IOCTL Include IP headers
+        self.ins.setsockopt(socket.SOL_IP, socket.IP_HDRINCL, 1)
+        self.outs.setsockopt(socket.SOL_IP, socket.IP_HDRINCL, 1)
+        # Bind on all ports
+        self.ins.bind((socket.gethostname(), 0))
```
[the remark this answers is not in the document]
Comment (gpotter2) [reply scapy/arch/windows/native.py]: No it won't work on Windows :/

**Comment by gpotter2** [comment]
Basic DEMO code:
```
conf.use_winpcapy = False
a = conf.L3socket()
a.tshark()
```

**Comment by gpotter2** [comment]
I still need to:
- add tests
- handle properly the warnings when Winpcap isn’t available

**Comment by gpotter2** [comment]
This PR is going to extend a bit. I’m planning to use iphlpapi to replace Powershell calls by C calls. That should speed up booting by a lot

**Comment by gpotter2** [comment]
@p-l- @guedou This is ready to be reviewed. That is a big PR, but it really improves Windows support.

File: test/windows.uts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+= Get current seq number
+
+seq = get_current_icmp_seq()
+assert seq > 0
+
+= Ping
+~ netaccess needs_root
+
+with conf.L3socket() as a:
+    answer = a.sr1(IP(dst="www.google.com", ttl=128)/ICMP(id=1, seq=seq)/"abcdefghijklmnopqrstuvwabcdefghi", timeout=1)
```
[the remark this answers is not in the document]
Comment (gpotter2) [reply test/windows.uts]: You can inject anything. Windows however has a special restriction for ICMP, which we bypass using this trick

**Comment by guedou** [comment]
In that case, it will be good to add a regular TCP or UDP test. I was unsure will reading the PR that sending something else than ICMP was possible.

**Comment by gpotter2** [comment]
@guedou I tried to add a DNS test, but as usual it works locally but not on appveyor -_-

**Comment by guedou** [comment]
Thanks a lot! Too bad that tests are lot working.

**Comment by gpotter2** [comment]
That’s probably related to the firewall on AppVeyor :/ I’ll try one day to hook scapy into Windows firewall, but for now that will do

**Comment by guedou** [comment]
@p-l- what is you call?

**Comment by gpotter2** [comment]
Apparently that didn’t fix https://github.com/secdev/scapy/pull/1759
Will look into it

**Comment by p-l-** [comment]
@gpotter2 which issue is not fixed? 1759 is this one!

**Comment by gpotter2** [comment]
Sorry I meant https://github.com/secdev/scapy/issues/1719. But this is ready

**Pull request merged**